### PR TITLE
[Hinty] Core typing: autorun/as_resolvers

### PR DESCRIPTION
**Pull request by gpotter2**
Types:
- autorun
- as_resolvers

### CODE REVIEW

File: scapy/as_resolvers.py [hunk 1]
```diff
@@ -13,29 +13,41 @@
...
+)
+
 
 class AS_resolver:
     server = None
-    options = "-k"
+    options = "-k"  # type: Optional[str]
 
     def __init__(self, server=None, port=43, options=None):
+        # type: (Optional[Any], int, Optional[Any]) -> None
```
Comment (p-l-) [inline scapy/as_resolvers.py:30]: I think `server` (this parameter and the attribute) should be `Optional[str]`

**Comment by gpotter2** [comment]
Merge?

File: scapy/as_resolvers.py [hunk 2]
```diff
@@ -13,29 +13,40 @@
...
+)
+
 
 class AS_resolver:
     server = None
-    options = "-k"
+    options = "-k"  # type: Optional[str]
 
     def __init__(self, server=None, port=43, options=None):
+        # type: (Optional[str], int, Optional[str]) -> None
```
Comment (guedou) [inline scapy/as_resolvers.py:29]: s/int/Optional[int]/ ?

**Review by guedou — CHANGES_REQUESTED** [review]

File: scapy/as_resolvers.py (hunk 2 above)
Comment (gpotter2) [reply scapy/as_resolvers.py:29]: `Optional` is an alias for `Union[..., None]`

**Review by guedou — APPROVED** [review]

File: scapy/as_resolvers.py (hunk 2 above)
Comment (guedou) [reply scapy/as_resolvers.py:29]: Thanks!

**Pull request merged**